webapp/security/views.py

```diff
@@ -607,6 +607,36 @@ def update_statuses(cve, data, packages, releases):
     return updated_statuses
 
 
+def delete_statuses(cve, data):
+    statuses_to_delete = (
+        db_session.query(Status).filter(Status.cve_id == cve.id).all()
+    )
+    statuses = cve.packages
+
+    for package_data in data.get("packages", []):
+        name = package_data["name"]
+
+        if not statuses[name]:
```

Comment:
Won't this result in a `KeyError` if the key is not there? I think you want `if name in statuses`?

Reply:
You are correct. My mistake.